java/src/org/openqa/selenium/firefox/AddHasExtensions.java

```diff
@@ -77,7 +84,23 @@ public String installExtension(Path path, Boolean temporary) {
 
         String encoded;
         try {
-          encoded = Base64.getEncoder().encodeToString(Files.readAllBytes(path));
+          if (Files.isDirectory(path)) {
+            Path extZip = Paths.get(path.getFileName().toString()+".zip");
+            ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(extZip.toFile()));
+            Files.walkFileTree(path, new SimpleFileVisitor<Path>() {
+              public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                zos.putNextEntry(new ZipEntry(path.relativize(file).toString()));
+                Files.copy(file, zos);
+                zos.closeEntry();
+                return FileVisitResult.CONTINUE;
+              }
+            });
+            zos.close();
```

Comment:
Could this be done in a Try with Resources block?
Maybe we can extract those lines for the unzipping into another method so it looks more readable.

Reply:
Incorporated the changes. Please check.